assets/app/scripts/directives/oscKeyValues.js

```diff
@@ -45,7 +45,6 @@ angular.module("openshiftConsole")
         $scope.value = null;
         $scope.form.$setPristine();
         $scope.form.$setUntouched();
-        $scope.form.$setValidity();
```

Comment:
See comment below, `$scope.form.$setValidity()` wipes out `form.$valid` and `form.$invalid` state.  Not sure why, seems like you would use it here.